common/src/main/java/org/apache/rocketmq/common/rebalance/AllocateMessageQueueSticky.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.rocketmq.common.rebalance;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeSet;
+import org.apache.rocketmq.common.AllocateMessageQueueStrategy;
+import org.apache.rocketmq.common.constant.LoggerName;
+import org.apache.rocketmq.common.message.MessageQueue;
+import org.apache.rocketmq.logging.InternalLogger;
+import org.apache.rocketmq.logging.InternalLoggerFactory;
+
+public class AllocateMessageQueueSticky implements AllocateMessageQueueStrategy {
+    private static final InternalLogger log = InternalLoggerFactory.getLogger(LoggerName.COMMON_LOGGER_NAME);
+
+    private final Map<String, List<MessageQueue>> messageQueueAllocation;
+
+    private final List<MessageQueue> unassignedQueues = new ArrayList<MessageQueue>();
+
+    public AllocateMessageQueueSticky(Map<String, List<MessageQueue>> messageQueueAllocation) {
+        this.messageQueueAllocation = messageQueueAllocation;
+    }
```

Comment:
> IMO, It's would be better not to upload LitePullConsumerRebalanceByBrokerReliabilityTest.java and PushConsumerRebalanceByBrokerReliabilityTest.java, because they are not examples and will mislead users.

You're right. Those two tests had been taken off from the commit.

> Why not just new a hashmap in the constructor instead of passing one ？

Fixed. It's just a careless mistake. Thx for pointing out.